learning-otel-go/internal/core/task/service.go

```diff
@@ -0,0 +1,83 @@
+package task
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type TaskService struct {
+	repo TaskRepository
+}
+
+func NewTaskService(repo TaskRepository) *TaskService {
+	return &TaskService{repo: repo}
+}
+
+func (s *TaskService) CreateTask(title, description string, concluded bool) error {
+	// Obter o tracer para este service
+	ctx := context.Background()
+	tracer := otel.Tracer("task-service")
+
+	// Criar um span para a operação
+	ctx, span := tracer.Start(ctx, "CreateTask", trace.WithAttributes(
+		attribute.String("task.title", title),
+	))
+	defer span.End()
+
+	id := uuid.New().String()
+	task := Task{
+		Id:          id,
+		Title:       title,
+		Description: description,
+		Concluded:   concluded,
+	}
+
+	// Adicionar ID ao span para correlacionar
+	span.SetAttributes(attribute.String("task.id", id))
+
+	return s.repo.Create(task)
+}
+
+func (s *TaskService) GetTasks() ([]Task, error) {
+	ctx := context.Background()
+	tracer := otel.Tracer("task-service")
+
+	ctx, span := tracer.Start(ctx, "GetTasks")
+	defer span.End()
+
+	tasks, err := s.repo.GetAll()
+	if err != nil {
+		return nil, err
+	}
+
+	span.SetAttributes(attribute.Int("task.count", len(tasks)))
+	return tasks, nil
+}
+
+func (s *TaskService) EditTask(id string, newTask Task) error {
+	ctx := context.Background()
+	tracer := otel.Tracer("task-service")
```

Comment:
Dúvida, vi na DOC do OTEL que eles sugerem criar uma função de Handler para capturar o tracing "automático" para todas as rotas, pelo oq entendi vira um wrapper da função de Handler do GO.

![image](https://github.com/user-attachments/assets/20020d02-38a3-4947-94df-b112f8b22fef)

https://opentelemetry.io/docs/languages/go/getting-started/#instrument-the-http-server

Será que não fica melhor doq controlar o tracing na mão aqui?